crates/ruff_linter/src/settings/types.rs

```diff
@@ -593,7 +603,7 @@ pub type IdentifierPattern = glob::Pattern;
 #[derive(Debug, Clone, CacheKey, Default)]
 pub struct PerFileIgnores {
     // Ordered as (absolute path matcher, basename matcher, rules)
-    ignores: Vec<(GlobMatcher, GlobMatcher, RuleSet)>,
+    ignores: Vec<(GlobMatcher, GlobMatcher, bool, RuleSet)>,
```

Comment:
I think this should be a named struct (`CompiledPerFileIgnore`? `PerFileIgnoreMatcher`?) with named fields rather than a tuple, but for ease of review I didn't make that change here; will push it as a separate PR (unless a reviewer suggests I shouldn't.)

Reply:
Agreed.